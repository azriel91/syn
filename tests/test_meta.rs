extern crate syn;

mod features;

#[macro_use]
mod macros;

use syn::parse::Parse;
use syn::{Meta, MetaList, MetaNameValue, NestedMeta};
use std::fmt::Debug;

#[test]
fn test_parse_meta_item_word() {
    let code = "hello";

    snapshot!(code as Meta);
}

#[test]
fn test_parse_meta_name_value() {
    test::<MetaNameValue>("foo = 5");
}

#[test]
fn test_parse_meta_name_value_with_path() {
    test::<MetaNameValue>("foo::bar = 5");
}

#[test]
fn test_parse_meta_name_value_with_keyword() {
    test::<MetaNameValue>("static = 5");
}

#[test]
fn test_parse_meta_name_value_with_keyword_and_path() {
    test::<MetaNameValue>("extern::static = 5");
}

#[test]
fn test_parse_meta_name_value_with_bool() {
    test::<MetaNameValue>("true = 5");
}

#[test]
fn test_parse_meta_item_list_lit() {
    test::<MetaList>("foo(5)");
}

#[test]
fn test_parse_meta_item_list_with_path() {
    test::<MetaList>("foo::bar(5)");
}

#[test]
fn test_parse_meta_item_multiple() {
    test::<MetaList>("foo(word, name = 5, list(name2 = 6), word2)");
}

#[test]
fn test_parse_meta_item_multiple_with_path() {
    test::<MetaList>("foo::bar(word, name = 5, list(name2 = 6), word2, path::true)");
}

#[test]
fn test_parse_nested_meta() {
    let code = "5";
    snapshot!(code as NestedMeta);

    let code = "list(name2 = 6)";
    snapshot!(code as NestedMeta);
}

fn test<T>(input: &str)
where
    T: Parse + Into<Meta> + Debug,
{
    let inner = snapshot!(input as T);
    let meta = snapshot!(input as Meta);

    assert_eq!(meta, inner.into());
}
