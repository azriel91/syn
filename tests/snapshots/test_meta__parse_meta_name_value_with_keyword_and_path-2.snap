---
created: "2019-03-17T21:23:40.462661Z"
creator: insta@0.7.1
source: tests/test_meta.rs
expression: syntax_tree
---
NameValue(
    MetaNameValue {
        path: Path {
            leading_colon: None,
            segments: [
                PathSegment {
                    ident: Ident(
                        extern
                    ),
                    arguments: None
                },
                Colon2,
                PathSegment {
                    ident: Ident(
                        static
                    ),
                    arguments: None
                }
            ]
        },
        eq_token: Eq,
        lit: Int(
            LitInt {
                token: Literal {
                    lit: 5
                }
            }
        )
    }
)
